### DISCUSSION TITLE
Project Board: Logical OR and AND operators for filtering view.

### DISCUSSION THREAD

**Discussion by YoDaMa** [community/community#16456] [Projects and Issues] [answered]
I would like a way to have a logical OR when filtering views so I can say for instance "show me items in the current sprint and any items that have no sprint currently assigned"

**Comment by vintprox** [comment]
OMG yes, please implement this...

**Comment by mariorod** [comment]
Thanks for the feedback. This is something **high in our backlog** and when we start implementation or have a firm delivery date we will update.

**Reply by michael-hawker** [reply]
Ah, this was also discussed in #15655.

I had a [simple suggestion for syntax here](https://github.com/github-community/community/discussions/15655#discussioncomment-2966427):

>(label:external label:technology),repo:external/reponame
>
> Building on the `,` as OR syntax available now, just adding parathesis could help a lot here as this would read as:
>
> "Contains Label `external` and `technology`, OR it comes from the external/reponame repo."

**Reply by michael-hawker** [reply]
Also, when doing a group, just being able to add independent filters per group could be useful as well to help make things easier to understand.

**Reply by Chinoman10** [reply]
So... more than a year later, is it still high on that backlog? 😬

**Reply by stoutgz** [reply]
🦗

**Reply by jkytomak** [reply]
@mariorod , how high is it now after almost 3 years?

**Reply by os1315** [reply]
Is anyone working on this?

**Reply by beingminimal** [reply]
> Thanks for the feedback. This is something **high in our backlog** and when we start implementation or have a firm delivery date we will update.

@mariorod 
It's an interesting question why GitHub might not be using Copilot Coding Agent to tackle its own long-pending features. While AI coding tools like Copilot are incredibly powerful, there are several reasons why they aren't a silver bullet for immediately resolving all outstanding development tasks, especially for a company like GitHub.

**Reply by Malix-Labs** [reply]
@mariorod :
> This is something high in our backlog and when we start implementation or have a firm delivery date we will update

Any update?

**Reply by ElliotKetchup** [reply]
update??

**Reply by lifeisafractal** [reply]
It's painful when a google search for a critical feature like this bringing you to an almost 4 year old issue with crickets from the product side. Would love to see this feature as it is pretty much table stakes for a project management tool like this.

**Reply by 0cjs** [reply]
I'm curious as to what "high in our backlog" means. What sort of progress have you made on this in more than three and a half years?

**Reply by isanchez-ng** [reply]
At this point, with all the recent advances in AI agents, it honestly seems more likely that you could build a custom backlog management platform for your team that is connected to Issues and PRs with the GitHub API than get this feature from them 😅

**Comment by yalegria** [comment]
Is this still on the works?

**Reply by mayur-lankeshwar** [reply]
Is it there?

**Comment by erwanor** [comment]
This would be useful.

**Comment by blubcow** [comment]
I would love to input a list of issue numbers related to the project as a filter.
It would be easy to add the OR operator like "#100,#101,#102" which already works for custom fields...

**Comment by isanchez-ng** [comment]
I'm also very interested on this! I would love to make a filter on my Github Project Roadmap that looks like:

![image](https://github.com/community/community/assets/81649278/42f9a2ba-f6d6-40c0-984b-7cdacd0a676b)

or maybe something similar to what I can do it on the pull request's filter section with the field "involves":
![image](https://github.com/community/community/assets/81649278/69475344-42a3-4ec3-93ab-5b83a4c2b5e8)

but it seems (from the actual [docs](https://docs.github.com/en/issues/planning-and-tracking-with-projects/customizing-views-in-your-project/filtering-projects#combining-filters)) that you can just make an OR operator **inside a field** and not **between fields** 😢

**Reply by s3rg1o7** [reply]
still here in (almost) 2026, waiting for this crucially basic feature :crossed_fingers:

**Reply by s3rg1o7** [reply]
@mariorod  any news?

**Comment by astronaute77** [comment]
Any chance we get this in 2024?

**Reply by she026** [reply]
@mariorod could you provide us a status. This functionality his highly needed. Also have you any plans for supporting a EPIC level for issues?

**Comment by beegan** [comment]
This is one of the most basic expectations of any filtering based UI. I can't believe this is not yet supported.

**Comment by dequejenn** [comment]
Adding support as something desperately needed with github project views. We add different labels to issues if in a shared monorepo versus one of our team repos, so our view needs to be able to see `(repo:xxxxxx AND label:111111) OR repo:yyyyyy`

**Comment by beegan** [comment]
My primary use case is for something like a label='project-name' or workstream='Project'. Not all issues get the right label added, or the right workstream attached when added to the board, and a combo view of both is required to not lose issues. 

What I end up doing as a workaround is that I'll create a view with all items on board, filter on one attribute, and use that filter to update the second attribute, but the obvious path is `label='project-name' || workstream='Project'`

**Comment by ggnall** [comment]
This is the crux of the issue!
>Not all issues get the right label added

In other project mgmt tools, issues are forced into a single project (or team), which is the primary grouping mechanism. Since GH issues have 0-to-many project assignments, they aren't reliably captured. So a project becomes a thing you  have to manually (and automatically with limited rules) curate.

**Comment by Malix-Labs** [comment]
This would be particularly useful with `assignee:@me | reviewers:@me` queries

**Reply by mannykoum** [reply]
our team needs the feature for exactly this reason 👍🏽

**Reply by tylercwood91** [reply]
yessss please

**Comment by defenestration** [comment]
`-sprint:<@current` is how I filter this right now. Which will show things with a `null` value for the sprint as well. This probably only applies to date fields though.

I think `sprint:@current,@none` or something similar would be more obvious and flexible.

**Reply by AbidKhairyAK** [reply]
thanks man! my project is a lot cleaner now!

**Comment by dpierangeli** [comment]
This would be very useful for us, since we can add custom fields to projects we need to filter the boards with various values from different fields.

**Comment by 0cjs** [comment]
I just started using Project boards for the first time, and almost immediately ran into the lack of this. Sigh. I too think that this is pretty basic functionality, and ought not still be absent almost two and a half years after this Issue was raised. (I am rather doubting it's high in the backlog as claimed.)

**Reply by erwanor** [reply]
Moved to using Asana, partly because of this, unfortunately.

**Comment by krystophny** [comment]
@mariorod we also immediately ran into this problem when using project boards. Any news on the issues?

**Comment by IDrumsey** [comment]
** is one of the biggest companies on planet earth -> takes years to implement a small and highly request change like this

**Comment by DrEntropy** [comment]
I suspect no one at github is seeing this, because there is no way they are that badly managed.

**Comment by zbrown001** [comment]
aint no way this discussion has been sitting here without an update for 967 days 🙄

edit: also they have already added this feature to issues ([docs](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/filtering-and-searching-issues-and-pull-requests#building-advanced-filters-for-issues))

**Comment by sutr90** [comment]
For  GitHub Enterprise Server 3.13.6 we can use comma, to separate the values: `milestone:"5.1.1","5.1.2"`

**Reply by cgrindel** [reply]
This works! Thank you.

**Comment by IDrumsey** [comment]
Do we need to reach out to github developers on twitter or something lol?

**Comment by she026** [comment]
This works on prosjekt beta, but only for issues under repos. But prosjekt beta is so much better that I recommend at everybody takes it in use. It has real sub issues and parent issues. You have to be org. owner in order to do it. You do it here: https://github.com/features/issues/signup

**Comment by taylorkaren** [comment]
Add me to the list of people who would like to see this implemented!

**Comment by robertoost** [comment]
This would be so useful for sprint planning!

**Comment by RCheesley** [comment]
+1 on this!

**Comment by ajoshi31** [comment]
Definitely need this, just in case need to check the common project for f/e and b/e repository and filter combined milestones

milestone:"Phase 1 : MVP release backend" + milestone:"Phase 1: MVP Release frontend"

milestone:"Phase 1 : MVP release backend" OR milestone:"Phase 1: MVP Release frontend"

**Reply by danielr** [reply]
This can actually be done using the comma-notation:
`milestone:"Phase 1 : MVP release backend","Phase 1: MVP Release frontend"`

Works as long as you want to match multiple values of the *same* field.

**Comment by heespi** [comment]
+1 on this, please!

**Reply by kurtsys-vintecc** [reply]
+1, please!

**Comment by beingminimal** [comment]
@ashtom @mariorod Please Please Please. Why can't we have all pending high value features released with the power of copilot? Please solve this. Thanks

**Comment by mariusops** [comment]
+1 please

**Comment by jucc** [comment]
Just to add a voice to the choir and reinforce this would be an **extremely relevant** feature to have! Our use case is we have a project where we track issues across multiple repos and teams and want to create views on it that combine filters - e.g. "repo: X OR label: Y OR assigned: Z"

**Comment by jefflbrauer** [comment]
We have moved an entire Federal IT organization to github projects and I will echo the pain associated with the lack of program management and road mapping capabilities of GHP, which includes the lack of an OR operator in issue filters. We need to be able to see a roadmap of issues across repositories in a gantt view where issues roll up to their parents and parents inherit durations. Having an OR operator is key to this end user need.

In short, we need Github to invest more in GH Projects and not siphon ALL your resources off to Copilot.
________________________________
From: Curt J. Sampson ***@***.***>
Sent: Wednesday, December 31, 2025 2:16 AM
To: community/community ***@***.***>
Cc: Brauer, Jeffrey (SENTINEL GROUP LLC) ***@***.***>; Manual ***@***.***>
Subject: [EXTERNAL] Re: [community/community] Project Board: Logical OR and AND operators for filtering view. (Discussion #16456)


I'm curious as to what "high in our backlog" means. What sort of progress have you made on this in more than three and a half years?

**Comment by beingminimal** [comment]
@jaredpalmer Any hope?

**Comment by phifuh** [comment]
Today i tried to hide pull requests, which are linked to an existing issue but show all pull requests which are not linked yet, just to find out that this feature isnt supported yet. 

Usecase: 
Some minor chores, just dont need an issue (its not worth the time to create and link it), so creating a fast pull request which is still showing up in the "sprint" view makes so much sense for use

**Comment by isanchez-ng** [comment]
Guys, I think it might be time to give up hope on this.
It’s been almost four years since the issue was raised, and there haven’t been any updates. With all the recent advances in AI agents, it honestly seems more likely that you could build a custom backlog management platform for your team that is connected to Issues and PRs with the GitHub API than get this feature from them 😅

**Reply by 65156** [reply]
agree...this is becoming absolutely beyond ridiculous now, we probably need to sacrifice a baby goat to get this across the line this year.

**Comment by sunnyi101** [comment]
👋 appreciate the continued feedback here!

We're actively looking at improving the filtering experience in Projects, including support for combining filters across different fields with OR logic. This is something our team is currently exploring, and while I can't share a specific timeline yet, it's on our radar and we understand how important this is for your workflows.

I'll update this thread as we make progress. Thanks for your patience.

**Reply by 0cjs** [reply]
I'm afraid we simply can't trust you when you say this. This has been "high in your backlog" for _four years_ now, so it seems apparent that you have in the past been merely trying to placate customers rather than having any serious commitment to addressing issues like this. It would have been far better  if you'd just said, "We're not going to add this" so that people who need this feature can move on to another vendor. (Well, far better for your customers; we know where your interests lie.)

If you want to start to rebuild trust, you might drop the, "Your call is important to us; we're working on this" responses, which at this point nobody will actually believe, and instead give an actual explanation of what the actual problem is (or was) that makes this so difficult that in _four years_ you've been apparently unable to make any progress on it, and how you intend to address that.

**Reply by sunnyi101** [reply]
@0cjs - https://github.com/orgs/community/discussions/16456#discussioncomment-17666569

**Reply by 0cjs** [reply]
@sunnyi101 I'd just like to note that your reply does not address any of my concerns. (I realise that perhaps you are not allowed to address my concerns.) That said, thank your managers for finally allowing this to get done.

**Comment by sunnyi101** [comment, marked as answer]
As promised this was shipped today! 
https://github.blog/changelog/2026-07-16-advanced-search-for-projects-is-generally-available/
